controllers/ArticleController.js

```diff
@@ -0,0 +1,174 @@
+import cloudinary from 'cloudinary';
+import Utilities from '../helpers/utilities';
+import { Article, User } from '../models';
+import createArticleHelper from '../helpers/createArticleHelper';
+
+/** Class representing controller for articles. */
+class ArticleController {
+  /**
+     * Create an article for a user
+     * @param {object} req - The request object
+     * @param {object} res - The response object sent to user
+     * @return {object} A object containing created articles.
+     */
+  static createArticle(req, res) {
+    const {
+      title, description, body, tagList, imageUrl,
+    } = req.body.article;
+
+    const { userId } = req;
+
+    const articleObject = {
+      title, description, body, tagList, imageUrl, userId
+    };
+    /** check if image was provided in the request
+     * upload the image to cloudinary, save the article
+     * with the cloudinary URL in database but if an error
+     * was encountered from cloudinary go ahead and create the article
+     */
+    if (imageUrl) {
+      return cloudinary.v2.uploader.upload(imageUrl, { tags: 'basic_sample' })
+        .then(image => createArticleHelper(res, articleObject, image.url))
+        .catch(() => createArticleHelper(res, articleObject));
+    }
+
+    /** if there no image was provided go ahead to create the article */
+    return createArticleHelper(res, articleObject);
+  }
+
+  /**
+   * get an article using slug as query parameter
+   * @param {object} req - request object
+   * @param {object} res - response object
+   * @returns {object} - the found article from database or error if not found
+   */
+  static getArticle(req, res) {
+    const { slug } = req.params;
+
+    return Article
+      .findOne({
+        where: { slug, },
+        include: [{
+          model: User,
+          attributes: { exclude: ['id', 'email', 'hashedPassword', 'createdAt', 'updatedAt'] }
+        }],
+        attributes: { exclude: ['id', 'userId'] }
+      })
+      .then((article) => {
+        /** if the article does not exist */
```

Comment:
use // for single line comment

Reply:
Why do you think so?  is there any impact it will have?